src/pages/settings/PreferencesPage.js

```diff
@@ -91,9 +107,20 @@ const PreferencesPage = ({priorityMode, user, translate}) => {
                             icon={() => <Icon src={DownArrow} />}
                         />
                     </View>
-                    <Text style={[styles.textLabel, styles.colorMuted]}>
+                    <Text style={[styles.textLabel, styles.colorMuted, styles.mb6]}>
                         {priorityModes[priorityMode].description}
                     </Text>
+                    <Text style={[styles.formLabel]} numberOfLines={1}>
+                        {translate('preferencesPage.language')}
+                    </Text>
+                    <View style={[styles.mb2]}>
+                        <Picker
+                            onChange={locale => Onyx.merge(ONYXKEYS.PREFERRED_LOCALE, locale)}
```

Comment:
Heads up for the future: the [Expensify.cash philosophy](https://github.com/Expensify/Expensify.cash#the-ui-layer) dictates that all Onyx data should be managed in `src/libs/actions`, and the UI layer should only call actions, not manipulate Onyx data directly.

Reply:
True, we should probably change this. Also, I think we should be saving this to an NVP, so it is shared across devices.

Reply:
> True, we should probably change this

Done [here](https://github.com/Expensify/Expensify.cash/pull/3638)

> we should be saving this to an NVP, so it is shared across devices.

Agree, issue [here](https://github.com/Expensify/Expensify/issues/167702). Seems like a good FP to me.